cmd/agent/main_windows.go

```diff
@@ -37,6 +52,28 @@ func main() {
 		}
 	}
 	defer log.Flush()
+	setConsoleCtrlHandler.Call(
+		syscall.NewCallback(func(controlType uint) uint {
+			var sigStr string
+			switch controlType {
+			case ctrlCEvent:
+				sigStr = "CTRL+C"
+			case ctrlBreakEvent:
+				sigStr = "CTRL+BREAK"
+			case ctrlCloseEvent:
+				sigStr = "CTRL+CLOSE"
+			case ctrlLogOffEvent:
+				sigStr = "CTRL+LOG_OFF"
+			case ctrlShutdownEvent:
+				sigStr = "CTRL+SHUTDOWN"
+			}
+			log.Infof("Received control event '%s', shutting down...", sigStr)
+			// signals.Stopper <- true
+			// Completely bypass the run command stop logic as it takes too long to stop
+			app.StopAgent()
+			os.Exit(0)
+			return 1
```

Comment:
Is this line actually reachable since we `os.Exit` before it? The only case I can think of is if `os.Exit` fails but I am not sure if that can happen. In any case I think it would be nice to have a comment about it.

Reply:
Makes sense, I would make a comment about it to make it clear.